subsys/net/lib/coap/coap.c

```diff
@@ -1397,3 +1397,18 @@ struct coap_observer *coap_find_observer_by_addr(
 
 	return NULL;
 }
+
+u16_t coap_next_id(void)
+{
+	static u16_t message_id;
+	static int seeded;
```

Comment:
As you mentioned, it would be better if this is atomic_t

Reply:
@jukkar Had a busy last week so apologies for the latency. It's pushed now.